### ISSUE TITLE
Widget substitution sometimes produces a double tweet button?

### ISSUE THREAD

**Issue by pde** [EFForg/privacybadger#788]
This page has twitter and github widgets:

https://jschr.github.io/textillate/

The first time I visit it with PBChrome, I get just one PB Tweet button.  But on the second and subsequent visits, I get a double PB Tweet button.

**Comment by cooperq** [comment]
duplicate of #402 maybe? Intrestingly I only see one tweet button even on subsequent visits.

**Comment by ghostwords** [comment]
I can reproduce. Previously: #553.

**Comment by rak-phillip** [comment]
So, I'm going to try and report on this issue as best I can.

When looking at the example from bug #553, there is a script after the twitter 
iframe that we want to replace:

```
!function(d,s,id){
  var js, fjs=d.getElementsByTagName(s)[0], p=/^http:/.test(d.location)?'http':'https';
  if(!d.getElementById(id)){
    js=d.createElement(s);
    js.id=id;
    js.src=p+'://platform.twitter.com/widgets.js';
    fjs.parentNode.insertBefore(js,fjs);
  }
}
(document, 'script', 'twitter-wjs');
```

It looks like this script generates a Twitter link if one does not exist and 
either:
1. Chrome is able to detect the `.twitter-share-button` class of the would be 
created button is replacing two buttons
2. Chrome is replacing the iframe with privacy badger button before the script 
evaluates; the script inserts the twitter link; we replace the newly created 
link with privacy badger button

I'm still trying to trace all the pieces and identify why Firefox only 
detects the iframe and Chrome is able to detect both the iframe and link that (as 
far as I know) shouldn't exist while our script is running.

**Comment by ghostwords** [comment]
Note that this is only a problem when the widget being replaced has been "cookieblocked". Blocked widgets are not affected since then there is no race between the widget's JavaScript and Privacy Badger's. Unblocked widgets no longer get replaced as of #1858.

**Comment by ghostwords** [comment]
Looks like what's happening is that Twitter's JavaScript creates an iframe and then removes the anchor element that's been defined as part of the Twitter button embed code. These elements both have a "twitter-share-button" class that Privacy Badger looks for.

On some page loads, Privacy Badger's content script gets two matches for `.twitter-share-button`, the iframe and the anchor, and replaces both. This happens when Twitter's JavaScript already created the iframe but hasn't yet removed the anchor.

As I wrote in my previous comment, this can only happen when Twitter is cookieblocked, as its JavaScript would not get to do anything when blocked.

**Comment by ghostwords** [comment]
Google Plus seems to be similarly affected: #402, #404.
